### ISSUE TITLE
Querying in a sub query seems not exclude deleted items

### ISSUE THREAD

**Issue by jeromelefeuvre** [makinacorpus/django-safedelete#148]
As you see below, items are not excluded....

358278 is soft deleted and not appear in a normal query
```
In [11]: WidgetItem.objects.values_list('widget').filter(id=358278)
Out[11]: <SafeDeleteQueryset []>
```
But not in a more complex query like this:
```
In [8]: widget_items = WidgetItem.objects.values_list('widget').filter(id=358278)

In [9]: Widget.objects.filter(id__in=widget_items).distinct()
Out[9]: <SafeDeleteQueryset [<Widget: 358278>]>
```
Any recommandations ?

And when printing query, deleted field are not excluded too:
```
In [8]: print(WidgetItem.objects.filter(id=358278).values_list('widget').query)
SELECT `nfb_widgets_widgetitem`.`widget_id` FROM `nfb_widgets_widgetitem` WHERE `nfb_widgets_widgetitem`.`id` = 358278 ORDER BY `nfb_widgets_widgetitem`.`order` ASC
```

Thanks!

**Comment by markui** [comment]
What is happening I'm also having this problem. Did this issue been existing from the past??

**Comment by AsheKR** [comment]
I would like to know that as well. Thanks

**Comment by Gagaro** [comment]
Thanks for the report,

I reproduced it in a test in the fix-subquery branch.

**Comment by Gagaro** [comment]
This is fixed in master, I'll have to migrate the test runners to github actions before releasing it.

**Comment by markui** [comment]
Thanks so much for your fast response. I look forward for the release soon..!!

**Comment by Gagaro** [comment]
released in 0.5.7
